sync/recursiveWatcher.go

```diff
@@ -13,7 +13,7 @@ import (
 )
 
 type RecursiveWatcher struct {
-	*fsnotify.Watcher
+	fsw     *fsnotify.Watcher
```

Comment:
this changes RecursiveWatcher's nature from is_a fsnotify.Watcher to has_a fsnotify.Watcher, denying access to fsnotify.Watcher.Events which should prevent future races.